### DISCUSSION TITLE
Get node status without OT CLI access

### DISCUSSION THREAD

**Discussion by agalliazzo** [openthread/openthread#10621] [Q&A] [answered]
Hi Guys,
I know that I can retrieve node and network information through the OT CLI (like with netdata show, meshdiag, etc) but, is there any way to check the network devices other than through CLI (for example, a coap endpoint on the leader)? 

I have several SED that I need to understand if they are still online, they wake up every 10s to poll the network but, due to energy constrains I don't want to ping them (devices runs on a CR2032 and, with some simulations, the battery last for 3years without polling data, but once I start to get data it drops to 2 years, and this is expected due to the fact that coap take more time to communicate than a simple network poll). 

It will be really interesting if I can poll the leader to have network data from the eth network side. Anyone has any idea?

Best 
Alessio

**Comment by jwhui** [comment, marked as answer]
For the Thread topology, only the immediate parent maintains state for a given SED. By design, no Thread device (including the Leader) maintains state for all other Thread devices in the network.

If you are using other protocols like SRP or DHCPv6 that do maintain state on some central device, then it should be possible to query the respective server (e.g. SRP Server) for the existing state.

**Comment by agalliazzo** [comment]
I'm using SRP indeed! Great suggestion! At the moment I use a long lease time (in the order of hours) due to old bugs (now fixed) on the STM32 implementation on OT. But I can make the lease time short (my idea is 5minutes) so, if I cannot resolve the name, the device is offline... Another option  will be to have an observable coap resource on the device that send data every 5m but I think that this second option will be less energy efficient... 
I will do a couple of test on both the solution, thanks a lot for your feedback as usual!
